clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -198,7 +209,35 @@ public void poll(long timeout) {
      * @param now current time in milliseconds
      */
     public void poll(long timeout, long now) {
-        poll(timeout, now, true);
+        lock.lock();
+        try {
+            // send all the requests we can send now
+            trySend(now);
+
+            // ensure we don't poll any longer than the deadline for
+            // the next scheduled task
+            client.poll(timeout, now);
+            maybeTriggerWakeup();
+
+            now = time.milliseconds();
+
+            // handle any disconnects by failing the active requests. note that disconnects must
+            // be checked immediately following poll since any subsequent call to client.ready()
+            // will reset the disconnect status
+            checkDisconnects(now);
```

Comment:
Suppose we have the following calls:

poll(5000)
wakeup()
poll(5000)

If we happen to get a disconnect or if we have any completed requests at the time that the wakeup occurs, we won't see the response handlers fired until after the next call to poll returns. So adding the call at the beginning of the call seems to make sense to me.